cli/tests/pcluster/config/test_validators.py

```diff
@@ -679,6 +679,76 @@ def _kms_key_stubber(mocker, boto3_stubber, kms_key_id, expected_message, num_ca
             "'per_unit_storage_throughput' can only be used when 'deployment_type = PERSISTENT_1'",
             0,
         ),
+        (
+            {"storage_capacity": 1200, "deployment_type": "PERSISTENT_1", "automatic_backup_retention_days": 2},
+            None,
+            None,
+            1,
+        ),
+        (
+            {
+                "storage_capacity": 1200,
+                "deployment_type": "PERSISTENT_1",
+                "automatic_backup_retention_days": 2,
+                "daily_automatic_backup_start_time": "03:00",
+                "copy_tags_to_backups": True,
+            },
+            None,
+            None,
+            1,
+        ),
+        (
+            {"automatic_backup_retention_days": 2, "deployment_type": "SCRATCH_1"},
+            None,
+            "FSx automatic backup features can be used only with 'PERSISTENT_1' file systems",
+            1,
+        ),
+        (
+            {"daily_automatic_backup_start_time": "03:00"},
+            None,
+            "'automatic_backup_retention_days' must be greater than 0 if \
+                'daily_automatic_backup_start_time' or 'copy_tags_to_backups' parameters are provided.",
+            1,
+        ),
+        (
+            {"copy_tags_to_backups": True},
+            None,
+            "'automatic_backup_retention_days' must be greater than 0 if \
+                'daily_automatic_backup_start_time' or 'copy_tags_to_backups' parameters are provided.",
+            1,
+        ),
+        (
+            {"daily_automatic_backup_start_time": "03:00", "copy_tags_to_backups": True},
+            None,
+            "'automatic_backup_retention_days' must be greater than 0 if \
+                'daily_automatic_backup_start_time' or 'copy_tags_to_backups' parameters are provided.",
+            1,
+        ),
+        (
+            {
+                "deployment_type": "PERSISTENT_1",
+                "automatic_backup_retention_days": 2,
+                "imported_file_chunk_size": 1024,
+                "export_path": "s3://test",
+                "import_path": "s3://test",
+                "storage_capacity": 1200,
+            },
+            {"Bucket": "test"},
+            "Backups cannot be created on S3-linked file systems",
+            0,
+        ),
+        (
+            {
+                "deployment_type": "PERSISTENT_1",
+                "automatic_backup_retention_days": 2,
+                "export_path": "s3://test",
+                "import_path": "s3://test",
+                "storage_capacity": 1200,
+            },
+            {"Bucket": "test"},
+            "Backups cannot be created on S3-linked file systems",
+            0,
+        ),
     ],
 )
 def test_fsx_validator(mocker, boto3_stubber, section_dict, bucket, expected_error, num_calls):
```

Comment:
Since we're going to be adding even more validation for this section in the next PRs, it's worth considering if there's a cleaner way to separate this validation into different functions (e.g., one validator for the backup parameters, one validator for the restore from backup parameter constraints, one validator for the deployment type constraints, etc.). But maybe since it's all intertwined keeping it all in the same function is the best way to do it.

Either way, just something to think about. Let's keep the current PR as is.